tests/test_repro.py

```diff
@@ -1027,3 +1028,49 @@ def test(self):
         shutil.rmtree(self.dvc.scm.dir)
         ret = main(['repro', self.file1_stage])
         self.assertEqual(ret, 0)
+
+
+class TestReproAlreadyCached(TestDvc):
+    def test_import(self):
+        stage = self.dvc.imp(self.BAR, 'imported')
+
+        os.unlink('imported')
+
+        patch_download = patch.object(stage.deps[0],
+                                     'download',
+                                     wraps=stage.deps[0].download)
+
+        patch_checkout = patch.object(stage.outs[0],
+                                     'checkout',
+                                     wraps=stage.outs[0].checkout)
+
+        with self.dvc.state as state, \
+             patch_download as mock_download, \
+             patch_checkout as mock_checkout:
+
+            stage.run()
+
+            mock_download.assert_not_called()
+            mock_checkout.assert_called_once()
+
+    def test_run(self):
+        stage = self.dvc.run(deps=[self.FOO],
+                             outs=[self.BAR],
+                             cmd='echo bar > bar')
```

Comment:
I don't think this would work on windows. I would use a python script instead, similar to the way we do it in other tests.

Reply:
Ah, I was wrong, it does indeed work on windows.